### DISCUSSION TITLE
Loading new thumbnail

### DISCUSSION THREAD

**Discussion by RolandTi** [shaarli/Shaarli#1899] [Q&A]
Hello,
I tried to get rid of blazy on my theme by using instead the "new" `loading="lazy"`.

But Shaarli is not generating any new thumb. The old ones are displayed. I had to use generate new ones through admin.

Here is my code : 

```
  <figure class="thumbnail">
        <a href="{$value.real_url}" aria-hidden="true">
        <img loading="lazy" 
         src="{$root_path}/{$value.thumbnail}#" 
        alt="{$value.title_html}" 
        width="{$thumbnails_width}"
         height="{$thumbnails_height}" />
    </a>
  </figure>
```

Any clue ? :-)

**Comment by ArthurHoaro** [comment]
Hi, sorry for the late answer.

There is some JS involved here. If you look into `linklist.html`, you can find the following line:

```
{if="$value.thumbnail === null"}data-async-thumbnail="1"{/if}
```

It means that if the thumbnail has never been retrieved, we flag the div with custom attribute. This attribute is then used in a JS file called `metadata.js`, to retrieve the thumbnail asynchronously and then update the `<img` tag with the proper thumbnail URL.

This was added to prevent web pages completely hanging for a long time while the thumbnails are being retrieved. It can be disabled with the setting: `general.enable_async_metadata` set to `false`. 

Note that with the proper JS file loaded, you code works just fine:

![image](https://user-images.githubusercontent.com/1962678/203857362-1b626f74-c0f7-4bb8-83e5-96a2d437cf5b.png)
![image](https://user-images.githubusercontent.com/1962678/203857385-db1ffce2-9d16-4df9-ad0c-abebad93e519.png)

**Reply by RolandTi** [reply]
Thanks ! I was looking in thumbnails.min.js and not metadata.min.js, and honestly it's a little bit to "hardcore" for me all that "building with brew process to make unreadable javascript". So I kept all the JS without touching it on my theme :smiley:
